forms/serializers/form.py

```diff
@@ -245,20 +239,77 @@ class Meta:
             ),
         ]
 
+    def entry_generator(self, entries, sections=None, fields=None):
+        if "sections" in entries:
+            yield from self.entry_generator(entries["sections"], sections=[entry for entry in entries["sections"]])
+        if "fields" in entries:
+            yield from self.entry_generator(entries["fields"], sections=sections, fields=[entry for entry in entries["fields"]])
+        if not isinstance(sections, list) and isinstance(fields, list):
+            for field in fields:
+                value_dict = entries[field]
+                yield field, sections, value_dict
+        if isinstance(sections, list):
```

Comment:
I guess you could strip the third 'if' out of this and just yield what's now in its scope in the end of the function by calling the last if like:
`if isinstance(sections., list) and not isinstance(fields, list)`

(Hard to be sure if this is corrertc by just watching the code, but there should be a pattern where you could strip this into 3 ifs and yield the "non-recurring yield" in the end without iffing.